### PR TITLE
VirtualService: use a double for mirror_percent

### PR DESCRIPTION
**Pull request by brianwolfe**
https://github.com/istio/api/pull/1079 added a `mirror_percent` field to VirtualService as an unsigned integer.

To support mirroring less than 1% of traffic to a destination, the
mirror_percent field should provide specification of percentages less
than 1. Envoy permits mirroring fractions as small as 1 in 1,000,000.
The Delay and Abort messages already encountered this issue and
solved it by using a Percent (== double) field called percentage instead
of an integer field called percent.

Here, we support small fractions using the same pattern, using
mirror_percentage in place of mirror_percent and deprecating the
mirror_percent field.

When reading mirror_percentage, we will need to handle rounding
of numbers near 1.0e-6 carefully because the floating point
representation will not be exact.

### CODE REVIEW

**Comment by brianwolfe** [comment]
Waiting to hear back about how my company wants to handle the CLAs.

**Comment by brianwolfe** [comment]
@googlebot I signed it!

**Comment by rshriram** [comment]
Open new pr please. Not seems stuck

**Comment by howardjohn** [comment]
@googlebot

**Comment by howardjohn** [comment]
@brianwolfe the bot claims you haven't signed the CLA still. Its looking for a CLA associated with your username and/or your company email (not sure if its either or both or something else). If you did signed it feel free to message me and I might be able to help

**Comment by brianwolfe** [comment]
There was a miscommunication (needed someone a little higher up in our management chain). I think it should work now.

**Comment by howardjohn** [comment]
@googlebot

**Comment by brianwolfe** [comment]
@googlebot I signed it!

**Comment by howardjohn** [comment]
I checked the internal db, says its not signed. Might just be some delay, I'll check back in a few minutes

**Comment by brianwolfe** [comment]
@googlebot

File: networking/v1alpha3/virtual_service.proto [hunk 1]
```diff
@@ -419,10 +419,15 @@ message HTTPRoute {
   // destination.
   Destination mirror = 9;
 
+  // Percentage of the traffic to be mirrored by the `mirror` field.
+  // Use of integer `mirror_percent` value is deprecated. Use the
+  // double `mirror_percentage` field instead
+  google.protobuf.UInt32Value mirror_percent = 18 [deprecated=true];
```
Comment (rshriram) [inline networking/v1alpha3/virtual_service.proto]: add a $hide_from_docs to make sure this doesn't appear in docs

**Comment by google-admin-test** [comment]
Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: **Please visit <https://cla.developers.google.com/> to sign.**

Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.

----

#### What to do if you already signed the CLA

##### Individual signers

*   It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).

##### Corporate signers

*   Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
*   The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
*   The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
		

ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fistio%2Fapi%2Fpull%2F1173) for more info**.

**Pull request merged**